### ISSUE TITLE
Operator doesn't detect config overrides apply

### ISSUE THREAD

**Issue by l0kix2** [ytsaurus/yt-k8s-operator#291]
Or maybe not in 100% of cases. Some research needed here.

**Comment by Krock21** [comment]
Reproduced it:
```
krock21@NB-FJ00LKG7CC dirac % kubectl diff -f overrides.yaml
diff -u -N /var/folders/dc/mvynbykn3t94kc7bjz1726m80000gn/T/LIVE-76230812/v1.ConfigMap.dirac.dirac-config-overrides /var/folders/dc/mvynbykn3t94kc7bjz1726m80000gn/T/MERGED-1524640721/v1.ConfigMap.dirac.dirac-config-overrides
--- /var/folders/dc/mvynbykn3t94kc7bjz1726m80000gn/T/LIVE-76230812/v1.ConfigMap.dirac.dirac-config-overrides	2024-07-11 15:33:49
+++ /var/folders/dc/mvynbykn3t94kc7bjz1726m80000gn/T/MERGED-1524640721/v1.ConfigMap.dirac.dirac-config-overrides	2024-07-11 15:33:49
@@ -556,9 +556,9 @@
                     "pool_trees" = [
                         "default";
                     ];
-                    "prefix" = "//sys/yql_agent/benchmarks_dq";
+                    "prefix" = "//sys/yql_agent/benchmarks_dq_delete";
                     "token_file" = "/usr/yql_agent_token";
-                    "upload_prefix" = "//sys/yql_agent/benchmarks_dq";
+                    "upload_prefix" = "//sys/yql_agent/benchmarks_dq_delete";
                     "upload_replication_factor" = 3;
                     "user" = "yql_agent";
                     "vanilla_job_file" = [
krock21@NB-FJ00LKG7CC dirac % kubectl apply -f overrides.yaml
configmap/dirac-config-overrides configured
krock21@NB-FJ00LKG7CC dirac % kubectl diff -f yt.yaml
krock21@NB-FJ00LKG7CC dirac % kubectl apply -f yt.yaml
ytsaurus.cluster.ytsaurus.tech/dirac unchanged
```

Operator did not see this change

I had to apply yt.yaml with diff for operator to notice it

**Comment by kozubaeff** [comment]
+1 Had the same problem
